### ISSUE TITLE
Minimise pane shortcut

### ISSUE THREAD

**Issue by MikeJohnPage** [rstudio/rstudio#6073]
I often use the show panes shortcuts (e.g., **Ctrl+3**: Move Focus to Help, **Ctrl+5**: Show Files) to _quickly_ access useful features of RStudio. After showing the desired pane, I am unable to _quickly_ minimise the same pane without using the mouse to click the relevant minimise button. I would like to be able to use the same shortcut that opened the pane to minimise it again so that my initial pane layout is maintained (without having to use a mouse).

**Comment by gtritchie** [comment]
I'm hoping to add a "minimize current pane" command as part of general keyboard focus improvements and better tracking of the currently active tab. See #6225

**Comment by francisbarton** [comment]
I'm going to piggyback my feature request onto this existing one because there are already so many FRs, and my wishlist is very similar to @MikeJohnPage 's here.

Here's my dream RStudio behaviour:
(imagine I have Source at top-left, Console at top-right, all other panes below, if that helps)

- Ctrl+2 focuses Console (and restores/expands if currently minimised)
- Ctrl+2 pressed again minimises Console
- Ctrl+Shift+2 maximises Console to the _entire height_ of that side of the window (full height, not full maximise - preferring height/"half-maximise" to full maximise could be an option somewhere?)
- Ctrl+Shift+2 pressed again un-maximises Console - back to normal size or minimised (returns to pre-maximised state). So it's a toggle.

Same behaviour for other panes.

tl;dr:
- keyboard shortcut for minimise current pane (same shortcut as focus pane - toggle)
- height-maximise-only keyboard shortcut option as an alternative to full maximise

That would be amazing 🙏🏼

**Comment by francisbarton** [comment]
See also https://github.com/rstudio/rstudio/issues/5498#issuecomment-577311258

**Comment by francisbarton** [comment]
Here's a slightly different use case, to provide further support for this functionality.

I have Source pane in the top left quadrant. Help pane (for example) and others, bottom left quadrant. Both "open", at "medium" height. I can use the "maximise" button on Source to make this full height, and thereby minimise Help.

Let's say I then want to switch regularly between Source and Help, and I'd ideally like each pane maximised when I switch to it. The aim here is to reduce the need for scrolling down to access text, and is helpful when working on a smaller size or resolution display.

I can switch to Help using Ctrl+F3 or using the View menu: it comes up medium height. I then maximise it, thereby minimising Source. If I then switch to Source (which was previously maximised before I switched to Help), eg using Ctrl+1, it is again medium height rather than maximised. So I maximise it. I then switch to Help again (Ctrl+F3) - this is also normal height again, not maximised.

I'd like to be able to set up RStudio so that when I switch to a pane, that pane would automatically be maximised (full height). Or at least a keyboard shortcut to quickly maximise the current pane, that could be used straight after switching, as an alternative to using the mouse to click the maximise/minimise icon.

_NB when I'm talking about maximise here I'm just talking about pane height, not as in the current "Zoom" (full height and width) behaviour._ Thanks again.
